### ISSUE TITLE
no MIR available ICE instead of an ambiguity error

### ISSUE THREAD

**Issue by tmiasko** [rust-lang/rust#115380]
I tried this code:

a.rs:

```rust
pub use a::*;
mod a {
    pub use a::*;
    pub use b::*;
    mod a {
        pub fn f() {}
    }
    mod b {
        pub fn f() {}
    }
}
```

b.rs:

```rust
pub fn main() {
    a::f();
}
```

```console
rustc a.rs --edition=2018 --crate-type=lib
rustc b.rs --edition=2018 --crate-type=lib -L. --extern a
```

I expected it to fail with an ambiguity error, instead it ICEs.

The code compiles successfully in 1.72.

Since #113099 it generates an ICE (it seems that now when an ambiguity is detected, the processing is not fully finalized and effective visibilities and reachability is incomplete).

Reporting primarily due to the fact that the future incompatibility for ambiguous glob imports is not enabled by default for dependencies https://github.com/rust-lang/rust/issues/114095#issuecomment-1699059334, which hides the root cause of the issue.

### Error output

```
error: internal compiler error: compiler/rustc_monomorphize/src/collector.rs:963:9: no MIR available for DefId(20:8 ~ a[024e]::a::a::f)
```

**Comment by bvanjoi** [comment]
reduced: 

```rs
// issue-115380-extern.rs
pub use c::*;

mod c {
    pub use self::a::*;
    pub use self::b::*;
    mod a {
        pub fn f() {}
    }
    mod b {
        pub fn f() {}
    }
}

// issue-115380.rs
// compile-flags: --emit=link
// aux-build:issue-115380-extern.rs
extern crate issue_115380_extern;

fn main() {
    issue_115380_extern::f();
}
```

**Comment by bvanjoi** [comment]
thanks for the report, it will fix by https://github.com/rust-lang/rust/pull/115389
